api.go

```diff
@@ -0,0 +1,168 @@
+	UsableCPUs int           // Number of usbale logical CPU cores
+	HeapAlloc  uint64        // Number of bytes currently allocated on the heap
+	StackAlloc uint64        // Number of bytes currently allocated on the stack
+
+	KeyStoreLatency   time.Duration // The latency of the KES key store. Zero when the key store is not accessible
+	KeyStoreReachable bool          // Indicates whether the key store is reachable
+	KeystoreAvailable bool          // Indicates whether the key store is available and reachable
+}
+
+// MarshalJSON returns the State's JSON representation.
+func (s State) MarshalJSON() ([]byte, error) {
+	type JSON struct {
+		Version             string        `json:"version,omitempty"`
+		OS                  string        `json:"os,omitempty"`
+		Arch                string        `json:"arch,omitempty"`
+		UpTime              time.Duration `json:"uptime,omitempty"`
+		CPUs                int           `json:"num_cpu,omitempty"`
+		UsableCPUs          int           `json:"num_cpu_used,omitempty"`
+		HeapAlloc           uint64        `json:"mem_heap_used,omitempty"`
+		StackAlloc          uint64        `json:"mem_stack_used,omitempty"`
+		KeyStoreLatency     time.Duration `json:"keystore_latency,omitempty"`
+		KeyStoreUnreachable bool          `json:"keystore_unreachable,omitempty"`
+		KeyStoreUnavailable bool          `json:"keystore_unavailable,omitempty"`
+	}
+	if s.CPUs < 0 {
+		return nil, fmt.Errorf("kes: invalid number of CPUs '%d'", s.CPUs)
+	}
+	if s.UsableCPUs < 0 {
+		return nil, fmt.Errorf("kes: invalid number of usable CPUs '%d'", s.UsableCPUs)
+	}
+	if !s.KeyStoreReachable && s.KeystoreAvailable {
+		return nil, errors.New("kes: keystore is unreachable but available")
+	}
+	return json.Marshal(JSON{
+		Version:             s.Version,
+		OS:                  s.OS,
+		Arch:                s.Arch,
+		UpTime:              s.UpTime,
+		CPUs:                s.CPUs,
+		UsableCPUs:          s.UsableCPUs,
+		HeapAlloc:           s.HeapAlloc,
+		StackAlloc:          s.StackAlloc,
+		KeyStoreLatency:     s.KeyStoreLatency,
+		KeyStoreUnreachable: !s.KeyStoreReachable,
+		KeyStoreUnavailable: !s.KeystoreAvailable,
+	})
+}
+
+// UnmarshalJSON unmarshal the JSON data into State.
+func (s *State) UnmarshalJSON(data []byte) error {
+	type JSON struct {
+		Version             string        `json:"version"`
+		OS                  string        `json:"os"`
+		Arch                string        `json:"arch"`
+		UpTime              time.Duration `json:"uptime"`
+		CPUs                int           `json:"num_cpu"`
+		UsableCPUs          int           `json:"num_cpu_used"`
+		HeapAlloc           uint64        `json:"mem_heap_used"`
+		StackAlloc          uint64        `json:"mem_stack_used"`
+		KeyStoreLatency     time.Duration `json:"keystore_latency"`
+		KeyStoreUnreachable bool          `json:"keystore_unreachable"`
+		KeyStoreUnavailable bool          `json:"keystore_unavailable"`
```

Comment:
can we keep camelCased JSON tags for go styling reasons?

Reply:
is this because we already used snake cased fields and we need to follow this for backward compatibility reasons?